src/main/java/com/example/onlinebookstore/controller/BookController.java

```diff
@@ -31,7 +32,7 @@ public BookDto getBookById(@PathVariable Long id) {
     }
 
     @PostMapping
-    public BookDto createBook(@RequestBody CreateBookRequestDto requestDto) {
+    public BookDto createBook(@RequestBody @Valid CreateBookRequestDto requestDto) {
         return bookService.save(requestDto);
     }
 
```

Comment:
You can also add validation to @PathVariable. In this case add Validated annotation on your controller class

For example, 
`public void delete(@PathVariable @Positive Long id)`